engine/src/main/java/org/terasology/network/internal/ServerConnectListManager.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright 2017 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.network.internal;
+
+import com.google.gson.Gson;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.terasology.context.Context;
+import org.terasology.engine.paths.PathManager;
+import org.terasology.engine.subsystem.DisplayDevice;
+import org.terasology.i18n.TranslationSystem;
+
+import java.io.IOException;
+import java.io.Writer;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.Set;
+
+/**
+ * This class provides the methods needed to determine if a client is allowed to connect or not,
+ * based on the blacklist and whitelist files.
+ */
+
+public class ServerConnectListManager {
+
+    private static final Logger logger = LoggerFactory.getLogger(ServerConnectListManager.class);
+    private static final Gson gson = new Gson();
+
+    private Context context;
+    private Set blacklistedIDs;
```

Comment:
Just noticed [raw types](https://docs.oracle.com/javase/tutorial/java/generics/rawTypes.html) are used here and `@SuppressWarnings("unckecked")` is used when manipulating the sets. I think the reason you did it this way is because if you make them `Set<String>`, then you get the warning when loading from json.
However I would solve the problem the other way round, i.e. using `Set<String>` as the type for `blacklistedIDs` and `whitelistedIDs`, because in my understanding these sets will always contain `String` objects. Then we could happily remove the `SuppressWarnings` from `addToBlacklist` and `addToWhitelist`.

We would then get warnings in `loadLists()`. I can think of a couple of solutions:
* Simply adding `@SuppressWarnings("unckecked")` to `loadLists()` - in my opinion if we need to do something unchecked, then the proper moment to do it is when deserializing, and once we are done with this, the data in memory is properly typed
* (this may be an overkill) Instead of serializing two `Set`s, we could serialize a class which contains them. This way, the storage format of the lists would be a single json file with a structure like `{"blacklistedIDs": [...], "whitelistedIDs": [...]}`. The class could be
    * this class itself (`ServerConnectListManager`), however the name would not be very appropriate from a semantic point of view - maybe it could be renamed to something like `ServerConnectLists` - and proper measures should be taken to make sure other non-relevant fields are not serialized; I think the paths could be made static (should be the same for any instance, correct?), and if we decide to keep `context` we have to somehow mark it as non-serialized (I think it can be made with the `transient` modifier, but I'm not sure).
    * a separate class, to which the two `Set` fields and `isClientOnWhitelist`, `isClientOnBlacklist`, `getWhitelist` and `getBlacklist` methods could be moved

(sorry if we are bothering you with all these changes request, I think this one is quite important from a code quality point of view but didn't notice it yesterday - my bad)